src/features/search/use-search.ts

```diff
@@ -1,26 +1,53 @@
 import { useAtomValue } from 'jotai/utils';
 import { apiClientsState } from '@store/api-clients';
-import { FoundResult, NotFoundResult } from '@common/types/search-results';
+import {
+  BlockSearchResult,
+  FoundResult,
+  NotFoundResult,
+  SearchResultType,
+} from '@common/types/search-results';
 import { useQuery } from 'react-query';
 import { useAppSelector } from '@common/state/hooks';
 import { selectSearchTerm } from '@features/search/search-slice';
 import { useDebounce } from '@common/hooks/use-debounce';
+import { isNumeric } from '@common/utils';
+import { Block } from '@stacks/blockchain-api-client';
 
 export const useSearchQuery = (id: string) => {
-  const { searchApi } = useAtomValue(apiClientsState);
+  const { searchApi, blocksApi } = useAtomValue(apiClientsState);
   return useQuery(
     ['search', id],
     async () => {
+      let foundResult;
+      let notFoundResult;
+
       try {
-        const result = await searchApi.searchById({ id });
-        return result as FoundResult;
+        foundResult = await searchApi.searchById({ id });
       } catch (e) {
         try {
           const data = await e.json();
-          if (data && 'found' in data) return data as NotFoundResult;
-        } catch (e) {
-          return undefined;
+          if (data && 'found' in data) {
+            notFoundResult = data;
+          }
+        } catch (e) {}
+      }
+
+      try {
+        // Also check the block height
+        if (notFoundResult && isNumeric(id)) {
```

Comment:
maybe if it's numeric we skip `searchApi.searchById` and just fetch block by height?

Reply:
Good point, if it's numeric we know for sure it's not a hash